--- plugins/canola-core/linux/system.py.old	2010-01-06 18:15:04.000000000 +0300
+++ plugins/canola-core/linux/system.py	2010-01-06 18:15:57.000000000 +0300
@@ -111,7 +111,11 @@
     def __init__(self):
         ModelStatus.__init__(self, "Network")
         self.status = 0.0
-        self._setup()
+        try:
+            self._setup()
+        except:
+            self.status = 1.0
+
 
     def _setup(self):
         sbus = dbus.SystemBus()
