diff --git a/WebKit/efl/DefaultTheme/Makefile.am b/WebKit/efl/DefaultTheme/Makefile.am
index 964de43..912fd6c 100644
--- a/WebKit/efl/DefaultTheme/Makefile.am
+++ b/WebKit/efl/DefaultTheme/Makefile.am
@@ -7,12 +7,12 @@ EDJE_CC = edje_cc
 EDJE_FLAGS = -v -id $(top_srcdir)/$(THEME_DIR) 
 
 filesdir = $(datadir)/webkit-1.0/theme
-files_DATA = $(THEME_DIR)/default.edj
+files_DATA = $(THEME_DIR)default.edj
 
 EXTRA_DIST += \
 	$(THEME_DIR)default.edc 
 
-default.edj: Makefile $(EXTRA_DIST)
+$(THEME_DIR)default.edj: Makefile $(EXTRA_DIST)
 	mkdir -p $(THEME_DIR)
 	$(EDJE_CC) $(EDJE_FLAGS) \
 	$(top_srcdir)/$(THEME_DIR)default.edc \
