diff --git a/src/bin/Makefile.am b/src/bin/Makefile.am
index c8af6db..fe46518 100644
--- a/src/bin/Makefile.am
+++ b/src/bin/Makefile.am
@@ -10,7 +10,7 @@ main.c \
 test_snow.c \
 demo.h
 
-e_phys_demo_LDADD = $(top_builddir)/src/lib/libephys.la @EPHYS_LIBS@
-e_phys_demo_DEPENDENCIES = $(top_builddir)/src/lib/libephys.la
+e_phys_demo_LDADD = -lephys @EPHYS_LIBS@
+e_phys_demo_DEPENDENCIES = -lephys
 
 EXTRA_DIST = flake.png
